src/cnn/cnn.js

```diff
@@ -0,0 +1,81 @@
+const puppeteer = require("puppeteer");
+
+const {
+  makeAddArticlesAction,
+  makeUpdateArticleNormalContentAction,
+  makeUpdateArticleVideoContentAction
+} = require("../store/actions");
+const { makeArticlesWithoutContentSelector } = require("../store/selectors");
+const { store, saveStore } = require("../store/index");
+const { CNN } = require("../constant");
+const { complement, partition, or } = require("../utils");
+
+const { isCnnVideoArticle, isCnnStyleArticle } = require("./cnn-utils");
+
+const addArticles = makeAddArticlesAction(CNN);
+const updateArticle = makeUpdateArticleNormalContentAction(CNN);
+const updateArticleVideoContent = makeUpdateArticleVideoContentAction(CNN);
+const articlesWithoutContent = makeArticlesWithoutContentSelector(CNN);
+
+const CNN_URL = "https://www.cnn.com";
+
+const articleContentUpdates = async page => {
+  const title = await page.$eval("h1.pg-headline", title => title.innerHTML);
+  const authors = await page.$eval(
+    ".metadata .metadata__byline__author",
+    authors => authors.innerHTML
+  );
+  const timestamp = await page.$eval(
+    ".metadata .update-time",
+    timestamp => timestamp.innerText
+  );
+  const content = await page.$eval('[data-zn-id="body-text"]', body => {
+    const ps = body.querySelectorAll(".zn-body__paragraph");
+    return (
+      Array.from(ps)
+        // This is an advertisement...
+        .filter(p => !/^<a href=.*<\/a>$/.test(p.innerHTML))
+        .map(p => p.innerText)
+    );
+  });
+  return {
```

Comment:
I had a little bit of uncertainty with the content I wanted to collect before, which is why fox-news had some other items being collected. I think for now we can settle on just collecting these 4 items: title, authors, timestamp, content.

I think this fits well with the sentiment analysis aspect of this program. We can do cool stuff like
- Figure out news organizations general sentiment based on article keywords (content)
- Figure out particular authors general sentiment (as well as sentiment toward particular content)
- Align content/title/sentiment with the timing that those articles are posted

There's a lot of work there, as well as in trying to do simultaneous/scheduled collection of news. Aggregation of news (database? server?) as well. Yeah.